scripts/Dockerfile

```diff
@@ -21,17 +21,18 @@ ENV CMAKE_CXX_COMPILER=/usr/bin/g++
 ENV CMAKE_MAKE_PROGRAM=/usr/bin/make
 ENV LD_LIBRARY_PATH=/home/pybamm/.local/lib
 
-RUN conda create -n pybamm python=3.11
-RUN conda init --all
-SHELL ["conda", "run", "-n", "pybamm", "/bin/bash", "-c"]
-RUN conda install -y pip
+# Create a virtual environment
+ENV VIRTUAL_ENV=/home/pybamm/venv
+RUN python3.12 -m venv $VIRTUAL_ENV
+ENV PATH="$VIRTUAL_ENV/bin:$PATH"
```

Comment:
If we are going to use `uv`, could we let `uv` create the virtual environment using `uv venv`?

Reply:
For context, I am using [these](https://docs.astral.sh/uv/guides/integration/docker/#running-in-docker) procedures listed on `astral-uv` documentation to use `uv` inside the docker container in the latest commit.